Paws4Me/Paws4Me/Controllers/Logic/ViewController.swift

```diff
@@ -8,109 +8,243 @@
 import UIKit
 
 class ViewController: UIViewController, UISearchBarDelegate {
+    // MARK: - IBOulets
     @IBOutlet weak private var petTable: UITableView!
     @IBOutlet weak private var searchBar: UISearchBar!
-    private var filteredPetObject: AdoptPet?
-    private var adoptPetObject: AdoptPet?
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        getAdoption()
-        petTable.delegate = self
-        petTable.dataSource = self
-        self.title = "Adoptable Animals"
-        setUpSearchbar()
-    }
+    @IBOutlet weak var viewControllerTitle: UINavigationItem!
+    // MARK: - Vars/Lets
+       var searchBarController = UISearchBar()
+       var filteredPetObject: AdoptPet?
+       var adoptPetObject: AdoptPet?
+       var isSingleSearch = false
+       var indexSinglePet: Int?
+       var selectedGender: String = ""
+       var petType = ""
+
+    // MARK: - Life cycle
+       override func viewDidLoad() {
+           super.viewDidLoad()
+           getAdoption()
+           petTable.delegate = self
+           petTable.dataSource = self
+           viewControllerTitle.title =  "Adoptable Pets"
+           setUpSearchbar()
+       }
+
+    // MARK: - IBActions
+    @IBAction func btnCatTapped(_ sender: UIButton) {
+           petType = sender.titleLabel?.text?.lowercased() ?? ""
+           sender.layer.borderColor = UIColor.myAppPurple.cgColor
```

Comment:
You can move this code to your `UIButton` `extension` and then update the state with one line of code otherwise consistency will be hard to maintain within the app.

Reply:
Thank you for your suggestion. I will do so 👍🏻